app/actions/websocket/preferences.ts

```diff
@@ -32,16 +40,21 @@ export async function handlePreferencesChangedEvent(serverUrl: string, msg: WebS
     if (!operator) {
         return;
     }
-
     try {
         const preferences: PreferenceType[] = JSON.parse(msg.data.preferences);
         handleSavePostAdded(serverUrl, preferences);
+
         if (operator) {
             operator.handlePreferences({
                 prepareRecordsOnly: false,
                 preferences,
             });
         }
+
+        const shouldUpdate = await hasDisplayNamePrefChanged(serverUrl, preferences);
```

Comment:
this is checking after the change was made in the db, so how is this working ?

Reply:
At the very end of all the function calls, we have the `updateChannelsDisplayName` function from `app/local/channel` that queries the preference table to compute the 'newDisplayName'.  Hence, we would want to read the 'updated' db value.